pkg/sql/jobs_profiler_execution_details_test.go

```diff
@@ -224,8 +224,18 @@ func TestReadWriteProfilerExecutionDetails(t *testing.T) {
 		require.NoError(t, err)
 		continueCh <- struct{}{}
 		jobutils.WaitForJobToSucceed(t, runner, jobspb.JobID(importJobID))
-		require.True(t, strings.Contains(string(goroutines), fmt.Sprintf("labels: {\"foo\":\"bar\", \"job\":\"IMPORT id=%d\", \"n\":\"1\"}", importJobID)))
-		require.True(t, strings.Contains(string(goroutines), "github.com/cockroachdb/cockroach/pkg/sql_test.fakeExecResumer.Resume"))
+		expectedSubstrings := []string{
+			fmt.Sprintf("labels: {\"foo\":\"bar\", \"job\":\"IMPORT id=%d\", \"n\":\"1\"}", importJobID),
+			"github.com/cockroachdb/cockroach/pkg/sql_test.fakeExecResumer.Resume",
+		}
+		goroutinesStr := string(goroutines)
+		for _, substr := range expectedSubstrings {
+			if s.DeploymentMode().IsExternal() {
+				require.NotContains(t, goroutinesStr, substr)
```

Comment:
i believe these sad UX deficiencies relate to a bug in status server which i think y'all own? https://github.com/cockroachdb/cockroach/issues/117609

If so, before special casing this test, is it worth investigating this bug?

Reply:
@msbutler I’ve only excluded goroutines for external-process mode. It feels like they were left out on purpose for external-process mode. Since you guys know best from product point of view, do you know if that's the case?

Yeah that issue is on @cthumuluru-crdb's list. We will take a look.

Reply:
yeah, intuitively, I would think that a cockroach server running on a sql pod should be able to grab the goroutines on all relevant sql pods. But this seems like a harder problem to solve. No need to block this PR though.

Reply:
Yeah, I feel like the external case is a whole story by itself.